llvm/lib/Target/ARM/AsmParser/ARMAsmParser.cpp

```diff
@@ -7596,6 +7601,38 @@ bool ARMAsmParser::validateInstruction(MCInst &Inst,
 
   const unsigned Opcode = Inst.getOpcode();
   switch (Opcode) {
+  case ARM::VLLDM:
+    [[fallthrough]];
+  case ARM::VLLDM_T2:
+    [[fallthrough]];
+  case ARM::VLSTM:
+    [[fallthrough]];
+  case ARM::VLSTM_T2: {
+    // Since in some cases both T1 and T2 are valid, tablegen can not always
+    // pick the correct instruction.
+    if (Operands.size() == 4) { // a register list has been provided
```

Comment:
I think you will always have 4 operands, because you are doing the disassembly in `DecodeLazyLoadStoreMul`. This could be omitted, or made an assert.

Reply:
`DecodeLazyLoadStoreMul` is part of the Decoding process (Binary representation to MCInst), this code handle the Parsing (Assembly to MCInst).
When the assembly does not contain a registry list (e.g. legacy T1) there will be only 3 operands.